logical/sanitize.go

```diff
@@ -60,3 +65,46 @@ type HTTPWrapInfo struct {
 	CreationTime    time.Time `json:"creation_time"`
 	WrappedAccessor string    `json:"wrapped_accessor,omitempty"`
 }
+
+type HTTPSysInjector struct {
+	Response *HTTPResponse
+}
+
+func (h HTTPSysInjector) MarshalJSON() ([]byte, error) {
+	j, err := json.Marshal(h.Response)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fast path no data or empty data
+	if h.Response.Data == nil || len(h.Response.Data) == 0 {
+		return j, nil
+	}
+
+	// Marshaling a response will always be a JSON object, meaning it will
```

Comment:
List responses don't return a `[]string` -- they return an entry in the `Data` map with a key of `keys` whose value is a `[]string`.

Any code that is marshaling an `HTTPResponse` will always have an object at the top level.